### PR TITLE
Generate top 50 popular resources of AWS using 100 lines of code

### PR DESCRIPTION
**Pull request by AveryQi115**
Signed-off-by: Nicola115 <2225992901@qq.com>

### CODE REVIEW

**Review by zzxwill — APPROVED** [review]
/lgtm

**Review by wonderflow — COMMENTED** [review]
good job, we need to add English version as well

File: blog/2022-03-01-kubevela-generate-top-50-popular-resources-of-aws-using-100-lines-of-code.md [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+通过简单分析，我们发现 AWS 前 50 Terraform modules 的数据可以通过请求 [https://registry.terraform.io/v2/modules?filter%5Bprovider%5D=aws&include=latest-version&page%5Bsize%5D=50&page%5Bnumber%5D=1](https://registry.terraform.io/v2/modules?filter%5Bprovider%5D=aws&include=latest-version&page%5Bsize%5D=50&page%5Bnumber%5D=1) 获取。
+
+
+# 开始之前
+
+代码接受两个用户传入参数：
+
+* provider 的名称
+* 该provider对应的Terraform Modules 的 URL
```
Comment (BinaryHB0916) [inline blog/2022-03-01-kubevela-generate-top-50-popular-resources-of-aws-using-100-lines-of-code.md]: 所有中英文混排时，建议留一个空格。这样读者阅读起来更加友好。

**Comment by AveryQi115** [comment]
@BinaryHB0916 Hi, I'm just wondering where to put the english version file. Could you please tell me?

**Comment by BinaryHB0916** [comment]
> @BinaryHB0916 Hi, I'm just wondering where to put the english version file. Could you please tell me?

Sure, it locates at the Root directory: /blog

File: blog/2022-03-01-kubevela-generate-top-50-popular-resources-of-aws-using-100-lines-of-code.md [hunk 2]
```diff
@@ -0,0 +1,296 @@
+# 如何用 100 行代码快速引入 AWS 最受欢迎的 50 种云资源
+
+作者： **Avery Qi** （同济大学） 周正喜（阿里云)
+
+KubeVela 目前已经支持了 AWS，Azure, GCP, 阿里云，腾讯云、百度云、UCloud 等云厂商，也提供了[简单快捷的命令行工具](https://kubevela.io/docs/next/platform-engineers/components/component-terraform)引入云服务商的云资源，但是在 KubeVela 里一个一个地支持云服务商的云资源不利于快速满足用户对于云资源的需求，本文提供了一个方案，用不到 100 行代码快速引入 AWS 前 50 最受欢迎的云资源。
```
Comment (BinaryHB0916) [inline blog/2022-03-01-kubevela-generate-top-50-popular-resources-of-aws-using-100-lines-of-code.md]: 并列建议统一使用符号：、

**Comment by zzxwill** [comment]
> Sure, it locates at the Root directory: /blog

Then where to

File: blog/2022-03-01-kubevela-generate-top-50-popular-resources-of-aws-using-100-lines-of-code.md [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+
+同时，我们也期望用户受到本文的启发，贡献其他云服务商的云资源。
+
+
+# AWS 最受欢迎的云资源在哪里？
+
+Terraform 官网提供了各个云服务商的 Terraform modules，比如 AWS 的云资源 Terraform modules 是 [https://registry.terraform.io/namespaces/terraform-aws-modules](https://registry.terraform.io/namespaces/terraform-aws-modules)，云资源按照受欢迎的使用程度（下载量）排序，比如 AWS VPC 下载量为 1870 万次。
```
Comment (BinaryHB0916) [inline blog/2022-03-01-kubevela-generate-top-50-popular-resources-of-aws-using-100-lines-of-code.md]: 超链接建议直接引用：比如 [AWS 的云资源 Terraform modules](https://registry.terraform.io/namespaces/terraform-aws-modules)。

File: blog/2022-03-01-kubevela-generate-top-50-popular-resources-of-aws-using-100-lines-of-code.md [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+
+通过简单分析，我们发现 AWS 前 50 Terraform modules 的数据可以通过请求 [https://registry.terraform.io/v2/modules?filter%5Bprovider%5D=aws&include=latest-version&page%5Bsize%5D=50&page%5Bnumber%5D=1](https://registry.terraform.io/v2/modules?filter%5Bprovider%5D=aws&include=latest-version&page%5Bsize%5D=50&page%5Bnumber%5D=1) 获取。
```
Comment (BinaryHB0916) [inline blog/2022-03-01-kubevela-generate-top-50-popular-resources-of-aws-using-100-lines-of-code.md]: 超链接直接引用，同上。

File: blog/2022-03-01-kubevela-generate-top-50-popular-resources-of-aws-using-100-lines-of-code.md [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+* Source
+* Description
+* Verified
+
+结构体定义在结构体 `TFDownload `中，通过 http 库获取 json 数据，再通过 `json.Unmarshal` 解析出 Terraform modules 的结构体。
+
+
+## 批量生成云资源
+
+* 新建目录，生成资源所需文件
```
Comment (BinaryHB0916) [inline blog/2022-03-01-kubevela-generate-top-50-popular-resources-of-aws-using-100-lines-of-code.md]: 非并列关系的文字小节，建议使用序号 1 2 3 而不是 *

File: blog/2022-03-01-kubevela-generate-top-50-popular-resources-of-aws-using-100-lines-of-code.md (hunk 4 above)
Comment (zzxwill) [reply blog/2022-03-01-kubevela-generate-top-50-popular-resources-of-aws-using-100-lines-of-code.md]: 这里是为了突出 api url 的。

File: blog/2022-03-01-kubevela-generate-top-50-popular-resources-of-aws-using-100-lines-of-code.md (hunk 2 above)
Comment (zzxwill) [reply blog/2022-03-01-kubevela-generate-top-50-popular-resources-of-aws-using-100-lines-of-code.md]: 有道理～

**Comment by zzxwill** [comment]
> > @BinaryHB0916 Hi, I'm just wondering where to put the english version file. Could you please tell me?
> 
> Sure, it locates at the Root directory: /blog

中文文档怎么命名？放在哪里呢？

**Comment by BinaryHB0916** [comment]
> > > @BinaryHB0916 Hi, I'm just wondering where to put the english version file. Could you please tell me?
> > 
> > 
> > Sure, it locates at the Root directory: /blog
> 
> 中文文档怎么命名？放在哪里呢？

命令和英文一致。放在 /i18n/zh/docusaurus-plugin-content-blog

File: blog/2022-03-01-kubevela-generate-top-50-popular-resources-of-aws-using-100-lines-of-code.md [hunk 6]
```diff
@@ -0,0 +1,296 @@
+# 如何用 100 行代码快速引入 AWS 最受欢迎的 50 种云资源
```
Comment (wonderflow) [inline blog/2022-03-01-kubevela-generate-top-50-popular-resources-of-aws-using-100-lines-of-code.md]:
```suggestion
# KubeVela：如何用 100 行代码快速引入 AWS 最受欢迎的 50 种云资源
```

File: blog/2022-03-01-kubevela-generate-top-50-popular-resources-of-aws-using-100-lines-of-code.md (hunk 1 above)
Comment (AveryQi115) [reply blog/2022-03-01-kubevela-generate-top-50-popular-resources-of-aws-using-100-lines-of-code.md]: 好的，谢谢建议

File: blog/2022-03-01-kubevela-generate-top-50-popular-resources-of-aws-using-100-lines-of-code.md (hunk 4 above)
Comment (AveryQi115) [reply blog/2022-03-01-kubevela-generate-top-50-popular-resources-of-aws-using-100-lines-of-code.md]: 这里把超链接的URL直接列出来是因为用户需要提供相应URL作为参数，感觉暴露出来可以给用户展示一下URL的格式。

**Review by wonderflow — APPROVED** [review]
great job！

**Comment by zzxwill** [comment]
@BinaryHB0916 Can this blog be merged?

**Comment by BinaryHB0916** [comment]
> @BinaryHB0916 Can this blog be merged?

Sure, merged.

**Pull request merged**